### PR TITLE
Issue 1350 documenting email extractor plugin

### PR DESCRIPTION
**Pull request by angelomendonca**
#9

### CODE REVIEW

File: docs/email_extract_plugin_installation.rst [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+#.   After the initial installation process finishes, the adapter is staged and will be available once
+     the necessary  services are restarted. Click the **Restart Services** button on the screen.
+#.   Once the process is complete, the administrator is able to login to Soltra Edge and configure the
+     Email Extract Plugin from the top level Adapters menu. Before configuring please have a look at the :ref:`dependencies-ref`.
+
+
+Configuration
+-------------
+
+With the Email extraction plugin installed and firewall policy rules added(if needed), the Soltra Edge Administrator can configure
```
Comment (3rdpaw) [inline docs/email_extract_plugin_installation.rst]: add space between `added(if needed`

File: docs/email_extract_plugin_installation.rst [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+the an email account.
+
+See the :ref:`adapter-configuration-ref` section to properly configure the Email Extractor Plugin.
+
+
+.. _dependencies-ref:
+
+System Dependency
```
Comment (3rdpaw) [inline docs/email_extract_plugin_installation.rst]: Change to `Requirements`

File: docs/email_extract_plugin_installation.rst [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+-----------------
+
+.. important::
+
+    This is only required if you have an active running firewall. 
+
+
+Since the email extraction plugin connects to the email server to download emails you would need to add
```
Comment (3rdpaw) [inline docs/email_extract_plugin_installation.rst]: 1. Use consistent naming, you use "Email Extractor Plugin" (capitalized) in the rest of the document, so make sure all references use consistent formatting/capitalization.
2. This is one long run on sentence, either add add punctuation, reword, or both.

File: docs/email_extract_plugin_overview.rst [hunk 4]
```diff
@@ -0,0 +1,19 @@
+.. _overview:
+
+Overview
+==========
+
+Extract Data From Mailing Lists
+*******************************
+The Email Extractor Plugin for Soltra Edge provides the functionality to extract data from
```
Comment (3rdpaw) [inline docs/email_extract_plugin_overview.rst]: Consistently use `CTX/Soltra Edge` throughout document

File: docs/email_extract_plugin_overview.rst [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+mailing lists. A user can sieve out IP Addresses, Email Addresses, URLs and File Hashes (MD5, SHA1,SHA256).
```
Comment (3rdpaw) [inline docs/email_extract_plugin_overview.rst]: 1. Add spaces after commas (between SHA1 and SHA256)
2. Change `A user can sieve out` to `Currently, the Email Extractor Plugin supports extraction of the following types of data: `

File: docs/email_extract_plugin_overview.rst [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+Review Result Set
+*****************
+With the Email Extractor Plugin, one can review the result set of imported emails and reject false positives.
```
Comment (3rdpaw) [inline docs/email_extract_plugin_overview.rst]: Use consistent language, `one` is typically a third person reference, but other places in the doc use `you` (second person).  Change to be consistent throughout doc

File: docs/email_extract_plugin_overview.rst [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+Generate STIX Based on selected results
```
Comment (3rdpaw) [inline docs/email_extract_plugin_overview.rst]: Capitalize consistently.  Either capitalize the first letter of every word or don't, but either way be consistent.  In this case, the options are:
1. `Generate STIX Based on Selected Results`
or
2. `Generate STIX based on selected results`

I don't have a preference, but make sure all headings throughout the documents use a consistent format.

File: docs/email_extract_plugin_support.rst [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+Troubleshooting
+----------------
+
+.. _iptables-rules-ref:
+
+Firewall Policy Rules
+#####################
+
+	By default IMAP access is blocked on CTX/Soltra Edge. To use the Email Plugin/Adapter/Whatever, you will need to add exclusions to your firewall as appropriate.
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: lol, as much as I like `Email Plugin/Adapter/Whatever`, please change to be consistent with rest of the document, so `Email Extractor Plugin` is what seems to be used the most for reference

File: docs/email_extract_plugin_installation.rst (hunk 1 above)
Comment (guitarmanvt) [inline docs/email_extract_plugin_installation.rst]: Please use `Email Intel Mailing List Plug-in` when referring to this in English.
Please use the lowercase `plugin` only in code.
Thanks.

File: docs/email_extract_plugin_support.rst [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+----------------
+
+.. _iptables-rules-ref:
+
+Firewall Policy Rules
+#####################
+
+	By default IMAP access is blocked on CTX/Soltra Edge. To use the Email Plugin/Adapter/Whatever, you will need to add exclusions to your firewall as appropriate. 
+
+	If your email provider settings need a SSL port then you would need to add a policy rule that allows traffic on that port. If on the other hand your email provider settings needs a Non SSL Port then you would need to add a policy rule to allow traffic on that port.
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: This is somewhat redundant, I would reword to: 

> Depending on the configuration of your email service, the ports used may change.  Examples are provided below for the most common ports used, IMAP (143) and IMAPS (993).  Please contact your email service provider for specific configuration information regarding which ports are used for IMAP access.

File: docs/email_extract_plugin_support.rst [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+Firewall Policy Rules
+#####################
+
+	By default IMAP access is blocked on CTX/Soltra Edge. To use the Email Plugin/Adapter/Whatever, you will need to add exclusions to your firewall as appropriate. 
+
+	If your email provider settings need a SSL port then you would need to add a policy rule that allows traffic on that port. If on the other hand your email provider settings needs a Non SSL Port then you would need to add a policy rule to allow traffic on that port. 
+
+	Some examples to open a  SSL (Port 993) and a Non SSL(Port number 143) have been shown below. If you are using something other than IPTables, please contact your firewall administrator and request the appropriate ports be opened if necessary.
+
+* IMAP Protocol (Non SSL Port)
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: Please change to IMAP Protocol (Port 143)

File: docs/email_extract_plugin_support.rst [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+ 	
+.. code-block:: shell
+
+	sudo iptables -A INPUT -p tcp --dport 143 -m conntrack --ctstate ESTABLISHED -j ACCEPT
+	sudo iptables -A OUTPUT -p tcp --sport 143 -m conntrack --ctstate NEW, ESTABLISHED -j ACCEPT
+
+* IMAPS Protocol (SSl Port)
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: Please change to IMAPS Protocol (Port 993)

File: docs/email_extract_plugin_support.rst [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+.. _iptables-rules-ref:
+
+Firewall Policy Rules
+#####################
+
+	By default IMAP access is blocked on CTX/Soltra Edge. To use the Email Plugin/Adapter/Whatever, you will need to add exclusions to your firewall as appropriate. 
+
+	If your email provider settings need a SSL port then you would need to add a policy rule that allows traffic on that port. If on the other hand your email provider settings needs a Non SSL Port then you would need to add a policy rule to allow traffic on that port. 
+
+	Some examples to open a  SSL (Port 993) and a Non SSL(Port number 143) have been shown below. If you are using something other than IPTables, please contact your firewall administrator and request the appropriate ports be opened if necessary.
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: 1. Change `SSL` to `IMAPS`
2. Change `Non SSL(` to `IMAP (`
3. Change `Port number` to `Port `

File: docs/email_extract_plugin_support.rst [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+	sudo iptables -A OUTPUT -p tcp --sport 143 -m conntrack --ctstate NEW, ESTABLISHED -j ACCEPT
+
+* IMAPS Protocol (SSl Port)
+
+.. code-block:: shell
+
+	sudo iptables -A INPUT -p tcp --dport 993 -m conntrack --ctstate ESTABLISHED -j ACCEPT
+	sudo iptables -A OUTPUT -p tcp --sport 993 -m conntrack --ctstate NEW, ESTABLISHED -j ACCEPT
+
+* Custom IMAP Port other than number (993/143)
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: Change to `Custom IMAP/IMAPS Port`

File: docs/email_extract_plugin_support.rst [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+.. code-block:: shell
+
+	sudo iptables -A INPUT -p tcp --dport <PORT_NUMBER> -m conntrack --ctstate ESTABLISHED -j ACCEPT
```
Comment (3rdpaw) [inline docs/email_extract_plugin_support.rst]: Add a commented out line at the beginning of this code snippet saying something along the lines of this:

`# To open a custom port, replace <PORT_NUMBER> with the custom port number`

File: docs/email_extract_plugin_usage.rst [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+
+.. note:: The steps on this page assume the Email Extractor Plugin has been properly configured.  If this is not the case, please reference the :ref:`adapter-configuration-ref`, or if you need assistance reference the support documentation, :ref:`email_extract_plugin_support-ref`.
+
+Detailed Overview
+###################
+
+Importing Emails
+-----------------
+
+Click on the Email Extract Extractor from the top level Adapters menu. The webpage displaying *Email Data Extractor* and a dropdown menu to choose the mailbox to process will be displayed. If you have more than one email account configured then you can
```
Comment (3rdpaw) [inline docs/email_extract_plugin_usage.rst]: Remove the stuff about having more than one account.  The plugin was developed with that in mind, but multiple account support isn't currently implemented.

File: docs/email_extract_plugin_usage.rst [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+    :align: center
+    :scale: 50 %
+    :alt: Data Extractor
+    :figclass: align-center
+
+    Figure 1. Data Extractor Screen
+
+
+
+Depending on the speed of your connection you will need to refresh the browser to see the number of emails that you have set to be
```
Comment (3rdpaw) [inline docs/email_extract_plugin_usage.rst]: Maybe change to something like "Once processing begins, it may take a few moments for data to be processed and displayed.  Please refresh the page periodically to load new results."

File: docs/email_extract_plugin_usage.rst [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+    Figure 1. Data Extractor Screen
+
+
+
+Depending on the speed of your connection you will need to refresh the browser to see the number of emails that you have set to be 
+downloaded.
+
+Viewing Results
+---------------
+Once the emails have been downloaded you can view them in the view results screen.
```
Comment (3rdpaw) [inline docs/email_extract_plugin_usage.rst]: Change all references to `view results screen` to `*View Results* screen`, so it's easier to understand that it's a "thing" for users reading the docs

File: docs/email_extract_plugin_installation.rst (hunk 1 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_installation.rst]: Email Extractor Plugin is what we will be using

File: docs/email_extract_plugin_installation.rst (hunk 2 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_installation.rst]: 👍

File: docs/email_extract_plugin_usage.rst (hunk 17 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_usage.rst]: 👍

File: docs/email_extract_plugin_usage.rst (hunk 16 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_usage.rst]: 👍

File: docs/email_extract_plugin_usage.rst (hunk 15 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_usage.rst]: 👍

File: docs/email_extract_plugin_support.rst (hunk 13 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_support.rst]: 👍

File: docs/email_extract_plugin_support.rst (hunk 12 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_support.rst]: 👍

File: docs/email_extract_plugin_support.rst (hunk 11 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_support.rst]: 👍

File: docs/email_extract_plugin_support.rst (hunk 10 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_support.rst]: 👍

File: docs/email_extract_plugin_support.rst (hunk 9 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_support.rst]: 👍

File: docs/email_extract_plugin_support.rst (hunk 8 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_support.rst]: :) Email Extract Plugin 👍

File: docs/email_extract_plugin_overview.rst (hunk 7 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_overview.rst]: 👍

File: docs/email_extract_plugin_overview.rst (hunk 5 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_overview.rst]: 👍

File: docs/email_extract_plugin_overview.rst (hunk 4 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_overview.rst]: 👍

File: docs/email_extract_plugin_installation.rst (hunk 3 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_installation.rst]: 👍

File: docs/email_extract_plugin_overview.rst (hunk 6 above)
Comment (angelomendonca) [inline docs/email_extract_plugin_overview.rst]: 👍

File: docs/conf.py [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+# Grouping the document tree into LaTeX files. List of tuples
+# (source start file, target name, title,
+#  author, documentclass [howto, manual, or own class]).
+latex_documents = [
+    (master_doc, 'STIXAPI.tex', u'STIX API Documentation',
+     u'Soltra', 'manual'),
+]
+
+man_pages = [
+    (master_doc, 'stixapi', u'STIX API Documentation',
```
Comment (guitarmanvt) [inline docs/conf.py]: Is this correct? `STIX API` => `Email Extract` maybe?

Comment (3rdpaw) [inline docs/conf.py]: If we ever start building man pages or using texinfo then yes.  For now I would say no since we'll likely move away from RST/Sphinx going forward.
